### ISSUE TITLE
Git: view should also show committed changes

### ISSUE THREAD

**Issue by josh-tepper** [microsoft/vscode#18676]
I wish that the git sidepanel also showed commited changes when on a feature (non-master) branch (i.e., the subset of commits since the merge-base with some base branch, e.g., master).

This would require knowing the base branch.  The subset of committed changes to show would then be:
`git diff $(git merge-base <BASE BRANCH> <BRANCH>) <BRANCH>`

There could be a field where the user enters the base branch (default master).  Alternatively, vscode can simply fix the base to always be master--at the very least, a good first step.

Such a feature would help in tracking a features development from start to finish, not just since the last commit.

![screen shot 2017-01-17 at 12 22 32 pm](https://cloud.githubusercontent.com/assets/20231989/22031750/a753a90a-dcaf-11e6-8eee-89bef4743001.png)

- VSCode Version: 1.8.1
- OS Version: Mac OS X - 10.11.6

**Comment by joaomoreno** [comment]
I'll put this together with #68

**Comment by josh-tepper** [comment]
@joaomoreno,
Would you consider re-opening this?  I actually disagree with the duplicate for the following reason:

This is much simpler and easier to implement than the open ended goal of showing the git history of a file or branch.  That involves rendering a graph of commits (or at least listing them) where a user needs to be able to select to arbitrary commits to compare.  That involves an entirely new interface.

The request here is for something much simpler: I simply want to list per-file diffs, in a place where per-file diffs are already listed.  There would be almost no new UX work -- simply another heading for "committed" changes.  When a user would click on a file, they would be shown the diff using the same diff panel as when clicking on staged or working tree changes.  Further, the diff would be unambiguous (in contrast to comparing to arbitrary commits in a files history).  It would be the total diff for a file between HEAD and the merge-base with the fixed base branch (typically master).

One other reason that I'm advocating for this solution is that I actually think this simpler solution is better -- I'm already looking in that area for the diffs.  Now, I want to also be able to see the diffs since I started the branch.  Further, the more complex task of total git file history is already handled by plugins.

Let me know what you think either way.  Feel free to completely disagree!

**Comment by joaomoreno** [comment]
We'll keep it around, thanks for the persistence!

**Comment by josh-tepper** [comment]
Thank you :-)  I apologize for the persistence!  Did you mean to re-open?

**Comment by josh-tepper** [comment]
I just wanted to follow up on this.  Are you guys still considering this?  It's something I wish for often, and I feel that it would be much less complex/more appropriate than some of the broader git history requests.

**Comment by joaomoreno** [comment]
Still open: still considering.

**Comment by josh-tepper** [comment]
As we push toward July/August releases, I wanted to again bring this up.  I feel like this is very low hanging fruit and could significantly improve the end-to-end development workflow.

TL;DR: Show committed changes "since" a base branch (master) in the left hand git panel

**Comment by josh-tepper** [comment]
@joaomoreno -- see above.  What do you think?

**Comment by joaomoreno** [comment]
Have you checked out [GitLens](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens)? They support a similar scenario.

**Comment by josh-tepper** [comment]
I have -- I'm a GitLens user.  But I'm not sure I know what feature you're talking about.  There is a feature that adds *stashed* files to the *explorer* (not the git view).  But there is nothing that shows the committed files in the gitview, AFAIK.

There is also directory compare feature, but this also doesn't accomplish the intent of this issue.

This issue's intent is to somewhere (git view or otherwise) produce a list of all files which have diffs as a result of a branch's commit history, and that this list should be the starting point for exploring the diffs.

**Comment by josh-tepper** [comment]
@joaomoreno -- with the release 1.16, I again want to advocate for putting this on the roadmap.  What do you think?

On the other hand, if this is, unfortunately, something that you would rule out, we can close this issue.

Let me know!

**Comment by josh-tepper** [comment]
This extensions accomplishes basically what I had in mind:
https://marketplace.visualstudio.com/items?itemName=letmaik.git-tree-compare#review-details

However, one drawback is that this is in the explorer view.  If something like this was in the source control view and had a flat structure (instead of a tree structure), it would provide a much more consistent approach to seeing the entirety of a change.

**Comment by Zone-Ghost** [comment]
Seconding this... Would be incredibly useful to be able to compare all changes made since the branch was started.  I'm constantly flipping between feature branches and committing my work in each and having the ability to review all changes easily would be great.  Git-tree-compare works well enough, but having it native with all the other bells and whistles would be nice.

**Comment by joaomoreno** [comment]
Sorry @josh-tepper, as part of our cleanup efforts, I will have to close this and simply merge it with #68.
